src/EmployeeTracker/api/views.py

```diff
@@ -0,0 +1,170 @@
+    return (outTime - tIn)
+
+
+class ListEmployeeAttendant(ModelViewSet):
+    serializer_class = AttendantSerializer
+    queryset = Attendants.objects.all()
+
+    @action(['get'], detail=False, url_path="WorkingHour/(?P<t>[^/.]+)")
+    def WorkingHour(self, request, *args, **kwargs):
+        type = kwargs.get('t')
+
+        try:
+            if type == 'w':
+                weekDateFromNow = datetime.now().date() - timedelta(days=7)
+                xo = [self.makingFunction(user, user.attendant.filter(date__gte=weekDateFromNow), 'Weekly') for user in
+                      User.objects.all()]
+                return Response(xo)
+
+            elif type == 'q':
+                weekDateFromNow = datetime.now().date() - timedelta(days=91)
+                xo = [self.makingFunction(user, user.attendant.filter(date__gte=weekDateFromNow), 'quarter') for user in
+                      User.objects.all()]
+                return Response(xo)
+            elif type == 'y':
+                weekDateFromNow = datetime.now().date() - timedelta(days=365)
+                xo = [self.makingFunction(user, user.attendant.filter(date__gte=weekDateFromNow), 'yearly') for user in
+                      User.objects.all()]
+                return Response(xo)
+            else:
+                raise Exception("Determine the period")
+        except Exception as ex:
+            return Response(str(ex))
+
+    @action(['get'], detail=False)
+    def commingToLeavingForHoleTeam(self, request, *args, **kwargs):
+        try:
+            everyEmpWithWorkingHoure = [self.makingFunction(user, user.attendant, 'yearly') for user in
+                                        User.objects.all()]
+            everyEmpWithleavingHoure = [self.makingFunction(user, user.leaves, 'yearly') for user in
+                                        User.objects.all()]
+            sumWorkHour = 0
+            for emp in everyEmpWithWorkingHoure:
+                sumWorkHour += emp["workingHouresyearly"]
+            if sumWorkHour == 0:
+                raise Exception("This Team has no work hour ")
+            sumLeaveHour = 0
+            for emp in everyEmpWithleavingHoure:
+                sumLeaveHour += emp["workingHouresyearly"]
+
+            return Response(f"this team has an leaving of percent {sumLeaveHour // sumWorkHour}")
+
+        except Exception as ex:
+            return Response(str(ex))
+
+    def makingFunction(self, emp, listAttends, t):
+        list = [x for x in listAttends.values('inTime', 'outTime')]
+        outtime = ""
+        for i in range(len(list)):
+            if list[i]['outTime'] != " ":
+                outtime = list[i]['outTime']
+            else:
+                list[i]['outTime'] = outtime
+        workingHoures = sum([subtractTime(t['inTime'], t['outTime']).hour for t in list])
+        return {"EmpName": emp.username, f"workingHoures{t}": workingHoures}
+
+
+
+class CreateAttendantEmployee(CreateAPIView):
```

Comment:
1. I see that you are using the `CreateAPIView` but you are also overriding the post method, which makes the extending of `CreateAPIView` useless.
2. Why are you sending the employee ID as a URL param? I think you should send it in the post data so you can benefit from the create method in the `CreateAPIView`, it will help you handle all exceptions and responses, and will make the code better.

other small note, the class name should be named like this `AttendantEmployeeCreateView`

so the code will be just like this:

```suggestion
class AttendantEmployeeCreateView(CreateAPIView):
    serializer_class = AttendantSerializer
    queryset = Attendants.objects.all()
    permission_classes = [IsAuthenticated]
```